Sources/BranchSDK/BranchOpenRequest.m

```diff
@@ -47,7 +47,7 @@ - (id)initWithCallback:(callbackWithStatus)callback isInstall:(BOOL)isInstall {
 }
 
 - (void)makeRequest:(BNCServerInterface *)serverInterface key:(NSString *)key callback:(BNCServerCallback)callback {
-    BNCRequestFactory *factory = [[BNCRequestFactory alloc] initWithBranchKey:key];
+    BNCRequestFactory *factory = [[BNCRequestFactory alloc] initWithBranchKey:key UUID:self.requestUUID TimeStamp:self.requestCreationTimeStamp];
```

Comment:
Right because of the factory pattern we'll need to do this for all request classes. Is there a super class that you can initialize these values inside `init`?

Reply:
@gdeluna-branch I assumed, these should be initialized at the time object is created, to get accurate timestamps. So I have initialized them in -`init` of `BNCServerRequest` Base class of all request types.  Also for persistence I have kept them inside request class because BNCRequestFactory is a local variable here and we will loose these values when this scope ends.
I was thinking to pass object reference (BranchOpenRequest *) in the future, when we will move things from pref. helper to request objects.